### ISSUE TITLE
Libdoc: Replace special `XML:HTML` format with dedicated `--specdocformat` option to control documentation format in spec files

### ISSUE THREAD

**Issue by Snooz82** [robotframework/robotframework#3731]
At the Moment the output `format` and the `documentation format of that spec file` are set by the same CLI option.

```text
 -f --format HTML|XML|XML:HTML
                          Specifies whether to generate an HTML or XML output
                          file. The XML:HTML format means generating an XML
                          output file where keyword documentation is converted
                          to HTML regardless of the original documentation
                          format. The default output format is got from the
                          output file extension so that `*.html` -> HTML,
                          `*.xml` -> XML and `*.libspec` -> XML:HTML.
```

That should be separated due to more format coming. starting with json.

New Feature would be to support an option `--specdocformat` `-s`
That option can have the values `RAW` or `HTML`.

When HTML is chosen the docs are converted to HTML docs in the export and docformat is set to 'HTML' 

That change is backwards incompatible due to the removal of `XML:HTML` 
However, the export to `*.libspec` stays as is.

instead of using this:
```
python -m robot.libdoc -f XML:HTML SeleniumLibrary SL.xml
```

use this:
```
python -m robot.libdoc -f XML -s HTML SeleniumLibrary SL.xml
```

**Comment by pekkaklarck** [comment]
This has been implemented by PR #3732 but it had some possible enhancements, especially related to docs, that I hope would still be addressed before closing this issue.

**Comment by Snooz82** [comment]
doc enhancements made by Peke
